docs/content/concepts/ops-jobs-graphs/graphs.mdx

```diff
@@ -262,6 +263,36 @@ Note that in most cases, it is usually possible to pass some data dependency. In
 
 Dagster also provides more advanced abstractions to handle dependencies and IO. If you find that you are finding it difficult to model data dependencies when using external storage, check out [IO managers](/concepts/io-management/io-managers).
 
+### Loading an asset as an input
+
+You can supply an asset as an input to one of the ops in a graph. Dagster can then use the [IO manager](/concepts/io-management/io-managers) on the asset to load the input value for the op.
+
+```python file=/guides/dagster/assets_ops_graphs/op_graph_asset_input.py
+from dagster import asset, job, op
+
+
+@asset
+def emails_to_send():
+    ...
+
+
+@op
+def send_emails(emails) -> None:
+    ...
+
+
+@job
+def send_emails_job():
+    send_emails(emails_to_send.to_source_asset())
+```
+
+We must use the <PyObject object="AssetsDefinition" method="to_source_asset" />, because <PyObject object="SourceAsset" plural /> are used to represent assets that other assets or jobs depend on, in settings where they won't be materialized themselves.
+
+If the asset is partitioned, then:
+
+- If the job is partitioned, the corresponding partition of the asset will be loaded.
+- If the job is not partitioned, then all partitions of the asset will be loaded.
```

Comment:
What does "all partitions will be loaded" mean for the shape of the value? Is it a list, or a dictionary, or a generator, or something else? I'm wondering how I ought to write my op to handle that.

Reply:
The type depends on the I/O manager implementation:
- The Pandas and PySpark type handlers of the DB IO managers (Snowflake, DuckDB, BigQuery) always return a single DataFrame, which can includes values from all the partitions.
- When loading an input that corresponds to multiple partitions, the UPathIOManager returns a dictionary that maps each input partition key to the input value for that partition key.

This needs better docs, but I don't think this is the right place to put them.